### ISSUE TITLE
qwen3-vl 8b finetune卡住

### ISSUE THREAD

**Issue by GuoHaoren** [QwenLM/Qwen3-VL#1668]
你好，我调通了qwen3-vl 8b的finetune之后，5445条数据，预估训练55个小时，但是训练了13个iteration，就卡住了
{'loss': 2.3233, 'grad_norm': 45.48319625854492, 'learning_rate': 3.0487804878048784e-07, 'epoch': 0.0}                                                                                               
{'loss': 2.2931, 'grad_norm': 43.93152618408203, 'learning_rate': 3.6585365853658536e-07, 'epoch': 0.0}                                                                                               
{'loss': 1.7906, 'grad_norm': 35.46949005126953, 'learning_rate': 4.26829268292683e-07, 'epoch': 0.0}                                                                                                 
{'loss': 1.8648, 'grad_norm': 34.76895523071289, 'learning_rate': 4.878048780487805e-07, 'epoch': 0.0}                                                                                                
{'loss': 2.1132, 'grad_norm': 46.419769287109375, 'learning_rate': 5.487804878048781e-07, 'epoch': 0.0}                                                                                               
{'loss': 1.9977, 'grad_norm': 44.094024658203125, 'learning_rate': 6.097560975609757e-07, 'epoch': 0.0}                                                                                               
{'loss': 1.9997, 'grad_norm': 36.365966796875, 'learning_rate': 6.707317073170733e-07, 'epoch': 0.0}                                                                                                  
{'loss': 1.9092, 'grad_norm': 39.004032135009766, 'learning_rate': 7.317073170731707e-07, 'epoch': 0.0}                                                                                               
{'loss': 1.7507, 'grad_norm': 44.85152053833008, 'learning_rate': 7.926829268292684e-07, 'epoch': 0.0} 

下面是我的config：
    qwenvl/train/train_qwen.py \
    --deepspeed ./scripts/zero3.json \
    --model_name_or_path "${llm}" \
    --dataset_use ${datasets} \
    --data_flatten False \
    --data_packing False \
    --tune_mm_vision False \
    --tune_mm_mlp True \
    --tune_mm_llm True \
    --bf16 \
    --output_dir ${output_dir} \
    --num_train_epochs 1 \
    --per_device_train_batch_size 2 \
    --per_device_eval_batch_size 1 \
    --gradient_accumulation_steps 8 \
    --max_pixels 50176 \
    --min_pixels 50176 \
    --eval_strategy "no" \
    --save_strategy "steps" \
    --save_steps 2500\
    --save_total_limit 10 \
    --learning_rate ${lr} \
    --weight_decay 0 \
    --warmup_ratio 0.03 \
    --max_grad_norm 1 \
    --lr_scheduler_type "cosine" \
    --logging_steps 1 \
    --model_max_length 2048 \
    --gradient_checkpointing True \
    --dataloader_num_workers 4 \
    --run_name ${run_name} \
    --report_to "none"

**Comment by JJJYmmm** [comment]
you can use pystack to see where the finetune script is stunk.

**Comment by fangjieJin** [comment]
请问您解决了吗？

**Comment by GuoHaoren** [comment]
并没有……

获取 Outlook for iOS<https://aka.ms/o0ukef>
________________________________
发件人: 金方杰 ***@***.***>
发送时间: Wednesday, November 5, 2025 7:31:13 PM
收件人: QwenLM/Qwen3-VL ***@***.***>
抄送: Guo Haoren ***@***.***>; Author ***@***.***>
主题: Re: [QwenLM/Qwen3-VL] qwen3-vl 8b finetune卡住 (Issue #1668)

        - External Email -



[https://avatars.githubusercontent.com/u/101873696?s=20&v=4]fangjieJin left a comment (QwenLM/Qwen3-VL#1668)<https://github.com/QwenLM/Qwen3-VL/issues/1668#issuecomment-3490685151>

请问您解决了吗？

―

**Comment by cendelian** [comment]
> 并没有……
> 
> 获取 Outlook for iOS<https://aka.ms/o0ukef>
> […](#)

你好 请问全量微调8b-vl需要什么显存配置？8卡*96G是否可以？

**Comment by penghao-wu** [comment]
@GuoHaoren Are you training on a dataset consisting of both multimodal and text-only data? It might hang in such a case with DeepSpeed Zero3.

**Comment by GuoHaoren** [comment]
> [@GuoHaoren](https://github.com/GuoHaoren) Are you training on a dataset consisting of both multimodal and text-only data? It might hang in such a case with DeepSpeed Zero3.

确实是这样。。。所以。。。有什么解决方法么

**Comment by penghao-wu** [comment]
> > [@GuoHaoren](https://github.com/GuoHaoren) Are you training on a dataset consisting of both multimodal and text-only data? It might hang in such a case with DeepSpeed Zero3.
> 
> 确实是这样。。。所以。。。有什么解决方法么

如果显存够可以用DeepSpeed Zero2，或者用fsdp2 (不wrap vision部分)，或者对于纯text的data加dummy image (可以参考llama-factory)

**Comment by hellokhj** [comment]
> 你好，我调通了qwen3-vl 8b的finetune之后，5445条数据，预估训练55个小时，但是训练了13个iteration，就卡住了 {'loss': 2.3233, 'grad_norm': 45.48319625854492, 'learning_rate': 3.0487804878048784e-07, 'epoch': 0.0} {'loss': 2.2931, 'grad_norm': 43.93152618408203, 'learning_rate': 3.6585365853658536e-07, 'epoch': 0.0} {'loss': 1.7906, 'grad_norm': 35.46949005126953, 'learning_rate': 4.26829268292683e-07, 'epoch': 0.0} {'loss': 1.8648, 'grad_norm': 34.76895523071289, 'learning_rate': 4.878048780487805e-07, 'epoch': 0.0} {'loss': 2.1132, 'grad_norm': 46.419769287109375, 'learning_rate': 5.487804878048781e-07, 'epoch': 0.0} {'loss': 1.9977, 'grad_norm': 44.094024658203125, 'learning_rate': 6.097560975609757e-07, 'epoch': 0.0} {'loss': 1.9997, 'grad_norm': 36.365966796875, 'learning_rate': 6.707317073170733e-07, 'epoch': 0.0} {'loss': 1.9092, 'grad_norm': 39.004032135009766, 'learning_rate': 7.317073170731707e-07, 'epoch': 0.0} {'loss': 1.7507, 'grad_norm': 44.85152053833008, 'learning_rate': 7.926829268292684e-07, 'epoch': 0.0}
> 
> 下面是我的config： qwenvl/train/train_qwen.py --deepspeed ./scripts/zero3.json --model_name_or_path "${llm}" --dataset_use ${datasets} --data_flatten False --data_packing False --tune_mm_vision False --tune_mm_mlp True --tune_mm_llm True --bf16 --output_dir ${output_dir} --num_train_epochs 1 --per_device_train_batch_size 2 --per_device_eval_batch_size 1 --gradient_accumulation_steps 8 --max_pixels 50176 --min_pixels 50176 --eval_strategy "no" --save_strategy "steps" --save_steps 2500 --save_total_limit 10 --learning_rate ${lr} --weight_decay 0 --warmup_ratio 0.03 --max_grad_norm 1 --lr_scheduler_type "cosine" --logging_steps 1 --model_max_length 2048 --gradient_checkpointing True --dataloader_num_workers 4 --run_name ${run_name} --report_to "none"

你好 请问其他配置你有修改吗 我的运行脚本和你的一样 但是会报优化器和学习对不上的问题
